From: Corey Minyard <cminyard@mvista.com>
Subject: unix_thread: Remove the fd handler sets before it's set up
References: 
Patch-Mainline: 
Git-commit: d613d279dbce2d5e4594f6fed39653d83af0d99b
Git-repo: https://git.code.sf.net/p/openipmi/code.git

There were a couple of fd handler set before the fd was actually
configured.  With recent changes in the selector code that would result
in an assert.  The calls were unnecessary, so remove them.

Signed-off-by: Corey Minyard <cminyard@mvista.com>


Signed-off-by:  <trenn@suse.com>
diff --git a/unix/posix_thread_os_hnd.c b/unix/posix_thread_os_hnd.c
index 28c66898..cb315c5e 100644
--- a/unix/posix_thread_os_hnd.c
+++ b/unix/posix_thread_os_hnd.c
@@ -140,8 +140,6 @@ add_fd(os_handler_t       *handler,
     fd_data->data_ready = data_ready;
     fd_data->handler = handler;
     fd_data->freed = freed;
-    sel_set_fd_write_handler(posix_sel, fd, SEL_FD_HANDLER_DISABLED);
-    sel_set_fd_except_handler(posix_sel, fd, SEL_FD_HANDLER_DISABLED);
     rv = sel_set_fd_handlers(posix_sel, fd, fd_data, fd_handler, NULL, NULL,
 			     free_fd_data);
     if (rv) {
